internal/pkg/message/outbox.go

```diff
@@ -139,6 +140,44 @@ func (ob *Outbox) Send(ctx context.Context, from, to address.Address, value type
 	return signed.Cid()
 }
 
+// Send a signed message, retaining it in the outbound message queue.
+// If bcast is true, the publisher broadcasts the message to the network at the current block height.
+func (ob *Outbox) SignedSend(ctx context.Context, message string, bcast bool) (out cid.Cid, err error) {
+	defer func() {
+		if err != nil {
+			msgSendErrCt.Inc(ctx, 1)
+		}
+	}()
+
+	head := ob.chains.GetHead()
+
+	m := types.SignedMessage{}
+
+	b := []byte(message)
+	err = json.Unmarshal(b, &m)
+	if err != nil {
+		return
+	}
+	signed := &m
+
+	height, err := tipsetHeight(ob.chains, head)
```

Comment:
Factor the duplicated code from here to the end of the method into a helper function shared by `Send`.